src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -131,10 +133,70 @@ std::shared_ptr<uint32_t> GetCounterHolder(const char * name)
 namespace chip {
 namespace Dnssd {
 
+/**
+ * @brief Returns the domain name from a given hostname with domain.
+ *        The assumption here is that the hostname comprises of "hostnameWithoutDomain.<domain>."
+ *        The domainName returned from this API is "<domain>."
+ *
+ * @param[in] hostname The hostname with domain.
+ */
+std::string GetDomainFromHostName(const char * hostnameWithDomain)
+{
+    std::string hostname = std::string(hostnameWithDomain);
+
+    // Find the last occurence of '.'
+    size_t last_pos = hostname.find_last_of(".");
```

Comment:
This should always be the last character of the string though? If it isn't then something else is going wrong.

Reply:
Yes, exactly.  This is making some assumptions, but then acting as if those assumptions are not being made, which means it will fail in bad ways if those assumptions happen to be violated.

Reply:
added a check that last_pos should be equal to the string length.

Reply:
N/A. The API was changed to look for the first occurence of '.' after talking to Ted seems like the hostname with domain has hostname.<domainname> after the first '.'. I will follow up on this after checking the matter spec on how the hostname could look like in different variations and will make any fixes as required.

Reply:
filed - #32681